modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/io/BplusInnerIo.java

```diff
@@ -81,23 +84,24 @@ public final long getLeft(long pageAddr, int idx) {
     public final void setLeft(long pageAddr, int idx, long pageId) {
         assertPageType(pageAddr);
 
-        putLong(pageAddr, offset0(idx, SHIFT_LEFT), pageId);
+        writePartitionless(pageAddr + offset0(idx, SHIFT_LEFT), link(pageId, 0));
```

Comment:
I believe that having a `link(pageId, 0)` is wrong here.

Reply:
Fix it